builtins.md

```diff
@@ -30,6 +30,7 @@ arguments without concern for overflow or underflow.
 | Char(integer) | Interpret the integer argument as a Unicode code point and return the corresponding character.|
 |Repr(integer) | Return a base 10 string representation of the argument. No equivalent for real numbers is currently defined.|
 |Is\_missing(arg) | Return true if the argument has value `missing`. This is the only way to test for `missing`.|
+|Unique_id() | Return an identifier for the current row that is guaranteed to be unique for the current category. When called repeatedly for the same row, the generated value is not guaranteed to be the same as previously generated for that row, but is guaranteed to be unique compared to all previous calls for the category. If any values of parent or child data names of the present data name are present, this method will return `missing` to avoid accidentally creating associations.
```

Comment:
```suggestion
|Unique\_id(`data\_name`) | Return an identifier for the current row that is guaranteed to be unique in the category of the `data\_name` data item. When called repeatedly for the same row, the generated value is not guaranteed to be the same as previously generated for that row, but is guaranteed to be unique compared to all previous calls for the category. If any values of parent or child data items of the `data\_name` data item are present in the data set, this method will return `missing` to avoid accidentally creating incorrect associations.
```

Reply:
If we allow a data name as a parameter I think we can simplify. How does the following sound?

"Unique_id(<cat>.<obj>): generate an identifier that is unique amongst all values of `<cat>.<obj>`. If any values of parent or child data items of `<cat>.<obj>` are present in the data set, this method will return `missing` to avoid accidentally creating incorrect associations."

I don't think we need to specify that it might return a different value when called repeatedly on the same row, indeed, it must according to the above definition. Also, this definition produces a well-defined outcome when this function appears in category methods and detached `Function` methods, although I don't anticipate any need for it in those cases.

Reply:
The new suggested definition looks great to me.